apps/web/src/features/positions/hooks/useRefetch.ts

```diff
@@ -0,0 +1,80 @@
+import { useCallback } from 'react'
+import useChainId from '@/hooks/useChainId'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { useAppSelector } from '@/store'
+import { selectCurrency } from '@/store/settingsSlice'
+import { usePositionsGetPositionsV1Query } from '@safe-global/store/gateway/AUTO_GENERATED/positions'
+import { useBalancesGetBalancesV1Query } from '@safe-global/store/gateway/AUTO_GENERATED/balances'
+import { usePortfolioGetPortfolioV1Query } from '@safe-global/store/gateway/AUTO_GENERATED/portfolios'
+import { FEATURES } from '@safe-global/utils/utils/chains'
+import { useHasFeature } from '@/hooks/useChains'
+import { useTokenListSetting } from '@/hooks/loadables/useLoadBalances'
+import useIsPositionsFeatureEnabled from './useIsPositionsFeatureEnabled'
+
+/**
+ * Hook for refetching positions and balances data.
+ * Automatically selects the appropriate endpoint (portfolio or legacy) based on feature flags.
+ *
+ * @returns Object containing:
+ *   - `refetch`: Function to refetch all data (positions + balances)
+ *   - `refetchPositions`: Function to refetch positions only
+ *   - `shouldUsePortfolioEndpoint`: Boolean indicating if portfolio endpoint is active
+ */
+export const useRefetch = () => {
```

Comment:
The name is too generic, I would suggest `useRefetchPortfolio`.
And in the end it duplicates a lot of the main portfolio fetching hook...